packages/orbit-components/src/TextLink/__tests__/index.test.tsx

```diff
@@ -87,12 +87,18 @@ describe("TextLink", () => {
     expect(element).toHaveAttribute("target", "_blank");
   });
 
-  it("should be focusable and have button role", () => {
-    render(<TextLink>{title}</TextLink>);
+  it("should respect custom tabIndex when provided", () => {
+    const onClick = jest.fn();
+    render(
+      <TextLink onClick={onClick} tabIndex={-1}>
+        {title}
+      </TextLink>,
+    );
 
     const element = screen.getByText(title);
 
-    expect(element).toHaveAttribute("tabIndex", "0");
+    expect(element.tagName).toBe("BUTTON");
+    expect(element).toHaveAttribute("tabIndex", "-1");
     expect(element).toHaveAttribute("role", "button");
```

Comment:
If this test is about tabIndex, we don't need to assert the button role and element